File: java/kode program/Menu.java
// Menu.java
public abstract class Menu {
    // Atribut 'protected' agar bisa diakses oleh kelas turunan
    protected String nama;
    protected int harga;
    protected String kodeMenu;
    protected String deskripsi;

    public Menu(String nama, int harga, String kodeMenu, String deskripsi) {
        this.nama = nama;
        this.harga = harga;
        this.kodeMenu = kodeMenu;
        this.deskripsi = deskripsi;
    }
    
    // Getter methods untuk diakses oleh kelas Restoran
    public String getNama() { return nama; }
    public int getHarga() { return harga; }
    public String getKodeMenu() { return kodeMenu; }
    public String getDeskripsi() { return deskripsi; }

    // Method abstract untuk menampilkan detail, jika diperlukan di luar tabel
    public abstract void tampilkanDetail();
}

File: python/kode program/menu.py
# menu.py

class Menu:
    def __init__(self, nama, harga, kode_menu, deskripsi):
        self.nama = nama
        self.harga = harga
        self.kode_menu = kode_menu
        self.deskripsi = deskripsi

    def tampilkan(self):
        print(f"[{self.kode_menu}] {self.nama} - Rp{self.harga}")
        print(f"    Deskripsi: {self.deskripsi}")